src/main/java/org/mariadb/jdbc/internal/failover/impl/AuroraListener.java

```diff
@@ -114,6 +149,20 @@ public void reconnectFailedConnection(SearchFilter searchFilter) throws QueryExc
 
         resetOldsBlackListHosts();
 
+        // Initial connection to cluster - retrieve endpoints immediately
+        if (searchFilter.isInitialConnection() && urlParser.getHostAddresses().size() == 1 && getClusterHostAddress() != null) {
+            AuroraProtocol tempProtocol = AuroraProtocol.getNewProtocol(getProxy(), getUrlParser());
+            tempProtocol.setHostAddress(getClusterHostAddress());
+            try {
+                tempProtocol.connect();
+                retrieveAllEndpointsAndSet(tempProtocol);
+            } catch (QueryException e) {
```

Comment:
I understand this logic, but why would we not use this fresh socket, and prefer to wait 1-2 seconds more to create a new connection ? this socket can be use directly. The only difference is that  cluster host can be blacklisted, but since blacklist is only a "connecting order preference" (i.e. driver connect hosts not in blacklist first, and use blacklist after).